spec/datadog/profiling/stack_recorder_spec.rb

```diff
@@ -340,16 +353,16 @@ def sample_types_from(decoded_profile)
       end
     end
 
-    describe 'heap samples' do
+    describe 'heap samples and sizes' do
       let(:sample_rate) { 50 }
       let(:metric_values) do
         { 'cpu-time' => 101, 'cpu-samples' => 1, 'wall-time' => 789, 'alloc-samples' => sample_rate, 'timeline' => 42 }
       end
       let(:labels) { { 'label_a' => 'value_a', 'label_b' => 'value_b', 'state' => 'unknown' }.to_a }
 
       let(:a_string) { 'a beautiful string' }
-      let(:an_array) { (1..10).to_a }
-      let(:a_hash) { { 'a' => 1, 'b' => '2', 'c' => true } }
+      let(:an_array) { (1..100).to_a.compact }
```

Comment:
Minor: Huh, interesting, I would've expected Ruby to right-size the underlying array (e.g. not need the `compact`) when doing `#to_a` on a range.

Reply:
It might not be needed but you had suggested calling it would enforce it in a previous comment so I added it for good measure

Reply:
It's indeed harmless. Usually an array gets created it gets right-sized, and only later appends trigger the growth behavior (although I haven't looked at Range#to_a so I'm assuming it calls the right APIs to do that ;) )